### PR TITLE
Actionable sns proposal banners

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

We want to display different state banners to provide more information when:
- user is not logged in
- Sns has no ballots support by governance canister
- there is nothing to vote for in selected Sns.

# Changes

- New banner `ActionableProposalsNotSupported`
- Update SnsProposalList
  - rename `isActionable` to `actionableSelected`
  - new prop `includeBallots` to control not-supported banner visibility
  - remove `ListLoader` because skeletons are enough and there is no pagination
  - display banners
- reorganised parameters in `SnsProposals` component (`proposals` is generated sooner to not avoid sorting for actionable)

# Tests

- updated props in `SnsProposalList` and remove skeleton test (already available in `SnsProposals`)
- banners visibility in `SnsProposals`

# Todos

- [ ] Add entry to changelog (if necessary).
Not yet

# Screenshots

## Not supported
image

## Not sign-in
image

## No proposals (mocked)
image

### CODE REVIEW

File: frontend/src/lib/components/sns-proposals/SnsProposalsList.svelte [hunk 1]
```diff
@@ -44,23 +50,23 @@
...
-  {#if $ENABLE_VOTING_INDICATION && isActionable}
-    {#if proposals === undefined}
-      <!-- TODO(max): TBD SignIn vs No vs NotSupported -->
-      <LoadingProposals />
-    {:else if proposals.length === 0}
-      <!-- TODO(max): TBD custom screen -->
-      <NoProposals />
-    {:else}
-      <div in:fade data-tid="actionable-proposal-list">
-        <ListLoader loading={isNullish(proposals)}>
```
Comment (mstrasinskis) [inline frontend/src/lib/components/sns-proposals/SnsProposalsList.svelte:56]: ListLoader not needed since we display skeletons.

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts [hunk 2]
```diff
@@ -67,25 +72,13 @@ describe("SnsProposalsList", () => {
...
+        snsName: undefined,
+        actionableSelected: false,
         nsFunctions: [],
       },
     });
 
     expect(queryByTestId("no-proposals-msg")).toBeInTheDocument();
   });
-
-  describe("actionable proposals", () => {
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:78]: Why was this removed?

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts [hunk 3]
```diff
@@ -67,25 +72,13 @@ describe("SnsProposalsList", () => {
     const { queryByTestId } = render(SnsProposalsList, {
       props: {
         proposals: [],
-        isActionable: false,
+        includeBallots: false,
+        snsName: undefined,
+        actionableSelected: false,
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:77]: You're passing the same 3 parameters each time.
Don't you need some tests with different parameter values to test that they have the intended effect?

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts [hunk 4]
```diff
@@ -470,16 +479,85 @@ describe("SnsProposals", () => {
...
 
+      it('should display "Not signIn" banner', async () => {
+        vi.spyOn(authStore, "subscribe").mockImplementation(
+          mockAuthStoreNoIdentitySubscribe
+        );
+
+        const po = await renderComponent();
+        await selectActionableProposals(po);
+
+        expect(await po.getActionableSignInBanner().isPresent()).toBe(true);
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposals.spec.ts:490]: Can you test that it's not present when "actionable" is not selected and also not present when signed in?

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts [hunk 5]
```diff
@@ -470,16 +479,85 @@ describe("SnsProposals", () => {
...
+      it('should display "No actionable proposals" banner', async () => {
+        actionableSnsProposalsStore.set({
+          rootCanisterId,
+          proposals: [],
+          includeBallotsByCaller: true,
+        });
+        const po = await renderComponent();
+
+        await selectActionableProposals(po);
+        expect(await po.getActionableEmptyBanner().isPresent()).toBe(true);
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposals.spec.ts:515]: Can you also test that this is not present in the cases where it should not be present?

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts [hunk 6]
```diff
@@ -470,16 +479,85 @@ describe("SnsProposals", () => {
...
+      it('should display "Actionable not supported" banner', async () => {
+        actionableSnsProposalsStore.set({
+          rootCanisterId,
+          proposals: [],
+          includeBallotsByCaller: false,
+        });
+        const po = await renderComponent();
+
+        await selectActionableProposals(po);
+        expect(await po.getActionableNotSupportedBanner().isPresent()).toBe(
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposals.spec.ts:527]: Can you also test that this is not present in the cases where it should not be present?

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts [hunk 7]
```diff
@@ -470,16 +479,85 @@ describe("SnsProposals", () => {
...
+          true
+        );
+        expect(
+          await po.getActionableNotSupportedBanner().getTitleText()
+        ).toEqual("Catalyze doesn't yet support actionable proposals.");
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposals.spec.ts:532]: Can you make sure that this file explicitly sets an SNS name and that that name appear here?
Then it's clear where the name comes from.
If some data is checked by the test, that data should be specified in the same file.

File: frontend/src/lib/i18n/en.json [hunk 8]
```diff
@@ -382,6 +382,10 @@
     "title": "There are no actionable proposals you can vote for.",
     "text": "Check back later!"
   },
+  "actionable_proposals_not_supported": {
+    "title": "$snsName doesn't yet support actionable proposals.",
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/i18n/en.json:386]: Looks like there is no a real difference 🤔
image

File: frontend/src/lib/components/sns-proposals/SnsProposalsList.svelte [hunk 9]
```diff
@@ -44,23 +50,23 @@
...
-      </div>
-    {/if}
+  {#if $ENABLE_VOTING_INDICATION && actionableSelected}
+    <div in:fade data-tid="actionable-proposal-list">
+      {#if !$authSignedInStore}
+        <ActionableProposalsSignIn />
+      {:else if isNullish(proposals)}
+        <LoadingProposals />
+      {:else if includeBallots === false}
+        <ActionableProposalsNotSupported snsName={snsName ?? ""} />
```
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/sns-proposals/SnsProposalsList.svelte:60]: I haven't found a nice solution for this case that will worth implementing. This is done more for the TS, because the actionable can't be loaded before the aggregator.

[the remark this answers is not in the document]
Comment (dskloetd) [reply frontend/src/lib/components/sns-proposals/SnsProposalsList.svelte:60]: In this component the `snsName` is an exported prop. If it should never be `undefined` here, let's declare it as `string` here instead of `string | undefined` and make sure it's not undefined where the value is actually determined. That would make it clearer why it shouldn't be undefined.

[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/src/lib/components/sns-proposals/SnsProposalsList.svelte]: Like [this](https://github.com/dfinity/nns-dapp/pull/4638/files#diff-caa46ccadffef9a8f7e402bb41ae8edb59c564d80d2efa4c79b86b6111fef7f5R144)?

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:78]: It was replaces with https://github.com/dfinity/nns-dapp/pull/4638/files#diff-5cba8a4dd828cb1f58437fa21e76deaba00eac7bf7907136ae0dcf1958ed9916R487.

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposals.spec.ts:485]: Good idea, extended.

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts (hunk 7 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/pages/SnsProposals.spec.ts]: Done!

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts [hunk 10]
```diff
@@ -480,14 +471,17 @@ describe("SnsProposals", () => {
...
-          mockAuthStoreNoIdentitySubscribe
-        );
+        setNoIdentity();
 
         const po = await renderComponent();
         await selectActionableProposals(po);
 
         expect(await po.getActionableSignInBanner().isPresent()).toBe(true);
+
+        authStore.setForTesting(mockIdentity);
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposals.spec.ts]: This is the same as `resetIdentity()` isn't it?

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts (hunk 3 above)
Comment (dskloetd) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:77]: Did you address this?

Comment (mstrasinskis) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:91]: As discussed, the block for actionable proposals has been added.

**Review by dskloetd — APPROVED** [review]
Thanks!
